src/main/java/athena/task/TimeData.java

```diff
@@ -70,19 +76,24 @@ public Time(boolean isFlexible, LocalTime startTime, int duration, String deadli
      * @throws InvalidRecurrenceException   Exception thrown when user mistypes recurrence
      * @throws InvalidDeadlineException     Exception thrown when user mistypes deadline
      */
-    public Time(Boolean isFlexible, String startTime, String duration, String deadline, String recurrence)
-            throws TaskDuringSleepTimeException,
+    public TimeData(Boolean isFlexible, String startTime, String duration, String deadline, String recurrence)
+            throws TaskDuringSleepTimeException, DateTimeParseException, TaskTooLongException,
             InvalidRecurrenceException, InvalidDeadlineException {
-        setIsFlexible(isFlexible);
-        setDuration(duration);
-        setDeadline(deadline);
+        this.duration = Integer.parseInt(duration);
+        this.isFlexible = isFlexible;
+        this.deadline = deadline;
```

Comment:
Any reason why you changed this? I think it'd be better to call the `set` methods as it all follows a similar level of abstraction